core/src/main/java/org/ehcache/core/spi/service/ServiceFactory.java

```diff
@@ -16,15 +16,30 @@
 
 package org.ehcache.core.spi.service;
 
+import org.ehcache.spi.ServiceProvider;
 import org.ehcache.spi.service.Service;
 import org.ehcache.spi.service.ServiceCreationConfiguration;
 
 /**
- * @author Alex Snaps
+ * A factory abstraction that can create {@link Service} instances.
  */
 public interface ServiceFactory<T extends Service> {
 
+  /**
+   * Creates an instance of the service using the passed in {@link ServiceCreationConfiguration}.
+   * <P>
+   *   Note that the mandatory aspect of the configuration is a service implementation concern.
```

Comment:
What "mandatory aspect"? Maybe "Note that a null configuration may be supported or even required by a service implementation"?